core/modules/installer/installer.browser.pages.inc

```diff
@@ -0,0 +1,998 @@
+      break;
+    case 'enable':
+      backdrop_set_title(t("Enable modules"));
+      $content = installer_browser_installation_enable_page();
+      break;
+  }
+  return theme('installer_browser_install', array('current_task' => $op, 'main_content' => backdrop_render($content)));
+}
+
+/**
+ * Task page for the Select versions installation task.
+ *
+ * Shows a form where the user can select which versions to install for each
+ * project.
+ */
+function installer_browser_installation_select_versions_page() {
+  module_load_include('inc', 'installer', 'installer.browser');
+  // Show a form that lets the user select which version of the projects to install
+  $queued_projects = installer_browser_get_queued_projects();
+
+  return backdrop_get_form('installer_browser_installation_select_versions_form', $queued_projects);
+}
+
+/**
+ * Form builder for the select versions form.
+ *
+ * @param $projects
+ *   An array of projects to get the releases for.
+ */
+function installer_browser_installation_select_versions_form($form, &$form_state, $projects) {
+  module_load_include('inc', 'installer', 'installer.browser');
+  backdrop_add_js(backdrop_get_path('module', 'installer') . '/js/select_releases.js');
+
+  $zip_loaded = installer_browser_check_zip_loaded();
+
+  $form = array();
+
+  // First unset any old data.
+  unset($_SESSION['installer_browser_install_releases_list']);
+
+  $form['#tree'] = TRUE;
+
+  $form['releases-header'] = array(
+    '#type' => 'item',
+    '#markup' => t("You're about to install:"),
+  );
+
+  $form['releases'] = array();
+
+  foreach ($projects as $project) {
+    // Get the available releases for this project.
+    if (!$release_data = installer_browser_get_project_release_data($project)) {
+      backdrop_set_message(t('Could not fetch releases for project %project.',
+        array('%project' => $project['title'])), 'warning');
+      watchdog('project_browser', 'Could not fetch releases for project %project.',
+        array('%project' => $project['title']), WATCHDOG_ERROR);
+      installer_browser_install_queue_remove($project['name']);
+      continue;
+    }
+
+    // We use the update module to calculate the recommended version.
+    $project_data = array(
+      'existing_major' => 0,
+      'existing_version' => 0,
+      'install_type' => '',
+    );
+    module_load_include('inc', 'update', 'update.compare');
+    update_calculate_project_update_status($project_data, $release_data);
+
+    $releases_list = array();
+
+    foreach ($release_data['releases'] as $version => $release) {
+      $release_title = t("@project @version - @date", array(
+        '@project' => $project['title'],
+        '@version' => $release['version'],
+        '@date' => format_date($release['date'], 'custom', 'M j, Y'),
+        ));
+      if (isset($release['terms']['Release type']) AND !empty($release['terms']['Release type'])) {
+        $release_title .= " (" . implode(', ', $release['terms']['Release type']) . ")";
```

Comment:
Maybe adding a check_plain() here wouldn't hurt in the event that "Release type" was unsafe information from the project browser server.